docs/upgrading_to_iam_4.0.md

```diff
@@ -0,0 +1,130 @@
+# Upgrading to IAM 4.0
+
+The 4.0 release of IAM is a backward incompatible release.
+
+## Details
+
+The 3.0 release have added support for applying bindings to dynamic variable values. This functionality was relying on `count` which
+in the end turned out to result in inconsistent module behavior
+when doing the configuration updates.
+
+Here is an example of such *dynamic* configuration in 3.0
+which applies bindings to two projects.
+
+```hcl
+module "project_factory" {
+  count = 2
```

Comment:
Modules do not yet support `count`. :smile: These examples should be adjusted to be realistic.